src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -903,7 +903,15 @@ fn traits_implemented_by<'a>(
                 ty
             );
             // Fast path: if this is a primitive simple `==` will work
```

Comment:
To avoid this happening again by mistake:

```suggestion
            // Fast path: if this is a primitive simple `==` will work
            // NOTE: The `match` is required; see #92662.
```